kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/types/TypeMatchersTest.kt

```diff
@@ -66,6 +56,14 @@ class TypeMatchersTest : WordSpec() {
           arrayList.shouldNotBeInstanceOf<ArrayList<*>>()
         }
       }
+      
+      "Allow execution with a lambda" {
+        val list = arrayListOf(1, 2, 3)
+        
+        list.shouldBeInstanceOf<ArrayList<Int>> { it: ArrayList<Int> ->
```

Comment:
type was made explicit to enforce that this type won't change in the future, even though it's redundant for users